packages/core-mobile/app/new/features/activity/utils.ts

```diff
@@ -129,3 +129,13 @@ export function isNftTransaction(tx: TokenActivityTransaction): boolean {
 export function isSupportedNftChainId(chainId: number): boolean {
   return isAvalancheCChainId(chainId) || isEthereumChainId(chainId)
 }
+
+// To differentiate between swap and sent/received
+// we need to check if the from and to addresses are the same
+// as the from and to addresses of the token
+export function isUnknownSwap(tx: TokenActivityTransaction): boolean {
```

Comment:
```suggestion
export function isSwapTransaction(tx: TokenActivityTransaction): boolean {
```
It might not be necessary to include “unknown” in the function name.